### DISCUSSION TITLE
Need help with adding a parameter to app_commands.choices

### DISCUSSION THREAD

**Discussion by mattskinners** [Rapptz/discord.py#9754] [Q&A] [answered]
![image](https://github.com/Rapptz/discord.py/assets/159703525/3ae1cbfb-dc84-4f48-bce1-2b5aa9cf488f)
This is what my code looks like now. I want to add a parameter to the choice 'custom', so when user selects it another field pops up in which he can insert a date.
I want it to work like this: /calendar [user chooses option "Custom"] [another field pops out here for user to enter a date]
Visual representation:
![image](https://github.com/Rapptz/discord.py/assets/159703525/c809c4a8-ee8a-40c5-ad59-36df608e20ee)

I just can't figure it out. Any help is appreciated.

**Comment by tailoric** [comment, marked as answer]
This is not possible with slash commands currently. Discord doesn't have inter-parameter dependency rules that allows you to activate fields / making them required because of a choice inside another option...

You could handle it in code that it will tell the user an error when "Custom" was chosen without a date and have date as an optional parameter but inside the UI what you want is not possible as a discord limitation
